src/main/java/com/github/invictum/reportportal/SuiteStorage.java

```diff
@@ -48,16 +49,54 @@ public void suiteFinisher(String id, Runnable finisher) {
      */
     public void finalizeActive() {
         suites.forEachKey(Runtime.getRuntime().availableProcessors(), id -> {
-            SuiteMetadata meta = suites.remove(id);
+            SuiteMetadata meta = suites.get(id);
+            if (meta.failedTests.isEmpty()) {
+                suites.remove(id);
+            }
             meta.finisher.run();
         });
     }
 
+    /**
+     * Register id of failed test for specific suite by id. Should be called if test failed first time.
+     * Part of retries logic.
+     *
+     * @param suiteId id of suite where fail test was detected
+     * @param testId  id of failed test
+     */
+    public void addFail(String suiteId, String testId) {
+        SuiteMetadata meta = suites.get(suiteId);
+        meta.failedTests.add(testId);
+    }
+
+    /**
+     * This method need for check is test is retry. Part of retries logic.
+     *
+     * @param suiteId id of suite where fail test was detected
+     * @param testId  id of failed test
+     */
+    public boolean isFailAdded(String suiteId, String testId) {
```

Comment:
This method checks is test fail present in a storage, so I suggest to name it more precise, e. g. `isFailPresent`

Reply:
Fixed. Thanks.